### DISCUSSION TITLE
Single database

### DISCUSSION THREAD

**Discussion by Kh-Oleg** [fief-dev/fief#229] [Q&A]
Hello,

What do you think about keeping all Fief-related data in a single database? 

The current solution with the main db and a db per workspace has the following disadvantages with regards to DB administration:
1. Since Fief creates new databases, the DB_USER must be granted privileges for the whole DB server, rather than for a particular database.
2. Workspace database names are referenced from the main db. That means that I cannot easily create a full copy of everything what Fief needs under another name on the same server - I must edit the main database manually, to make it working.

From DBA point of view, I'd prefer if Fief would require a single, pre-created database with all privileges limited to this particular database.

**Comment by frankie567** [comment]
Yes, I agree that it poses some issues in such context. I mainly thought about Fief in a cloud environment with multiple workspaces.

I was thinking about an approach like this:

* The "main workspace" would be on the main DB, no matter what (with tables prefixed by `fief_`).
* By default, creating workspaces would be forbidden; which is suitable for self-hosted deployments
* If multiple workspaces are enabled, then those other workspaces would be created under another schema like it is today.

I would also like to revisit BYOD so external databases are created in the main schema (see #212).
